### DISCUSSION TITLE
How can I disable the automatic check on for "Add to Keychain" whenever I open a connection on CyberDuck for Mac?

### DISCUSSION THREAD

**Discussion by seberkuhn** [iterate-ch/cyberduck#16124] [Q&A]
I am using Cyberduck v. 8.8.2 for Mac (macOS Sonoma 14.5) but this problem has haunted me for many years: one of the main sftp servers I use requires 2-factor authentication, with a generated password for every login. So I do NOT want to save whatever I type into the "Password" field in Keychain, since it not only would be wrong the next time, but also would then automatically override my entering the correct one. I am one slip of the finger away from accidentally forgetting to uncheck that option manually - is there any way that it is checked off AUTOMATICALLY by default (so I have to opt in instead of opt out)? Thanks!

**Comment by dkocher** [comment]
Uncheck _Preferences → Connection → Use Keychain_.

**Reply by seberkuhn** [reply]
Thanks, but that would disable Keychain passwords for ALL my servers, wouldn't it? I only want to make it so that I have to AFFIRM if I want the password for a specific connection saved, instead of it being preselected as "Add to Keychain".

**Reply by tatze96** [reply]
Correct. Currently, you can choose to use Keychain for all connections or none. There's no way to choose selectively for bookmarks whether you want to save them in Keychain or not.

**Reply by seberkuhn** [reply]
Thanks for your response. Sounds like a feature request for Cyberduck! Actually, I found out it isn't quite as bad as I thought - when I do disable Keychain for CyberDuck, it STILL remembers any passwords PREVIOUSLY entered. This is both good (until my passwords change, I don't have to re-enter them every time I connect to a server) and bad (if I accidentally left the "Add to Keychain" button selected, the wrong password will still haunt me on successive login attempts - I actually have to go into Keychain, find the wrong password and remove it). I'll make do with this workaround for now...
